foreigndonors.php

```diff
@@ -195,21 +196,44 @@ function foreigndonors_civicrm_buildForm($formName, &$form) {
     $form->assign('domainId', $domainId);
     $form->assign('foreignPageId', $formId);
 
+    // As a safeguard against people completing NSW contrib forms through the wrong domain
+    // We're going to get the financial type of the page/event and if it starts with 'NSW'
+    // we know to inject the prohibited donor checkbox, regardless of domain
+    $finType = [];
+    if ($formName == 'CRM_Contribute_Form_Contribution_Main') {
+      $finType = \Civi\Api4\ContributionPage::get(FALSE)
+        ->addSelect('financial_type_id:label')
+        ->addWhere('id', '=', $formId)
+        ->setLimit(1)
+        ->execute();
+    }
+    else {
+      $finType = \Civi\Api4\Event::get()
+        ->addSelect('financial_type_id:label')
+        ->addWhere('id', '=', $formId)
+        ->setLimit(1)
+        ->execute();
+    }
+
     // Add the checkbox to the public form
     // Have to use different language for Queensland
+    // Inject a prohibited donor checkbox for NSW
+    // if the domain is NSW or the fin type starts with 'NSW'
     if ($domainId == 7) {
       $label = "I am an Australian Citizen or Permanent Resident, and not a QLD prohibited donor";
       $form->add('Checkbox', 'foreigndonor', ts('I am an Australian Citizen or Permanent Resident, and not a QLD prohibited donor'));
       $form->addRule('foreigndonor', ts('You must affirm you are not a prohibited donor as per State and Federal legislation'), 'required', NULL, 'client');
     }
-    elseif ($domainId == 8) {
+    elseif ($domainId == 8 || substr($finType[0]['financial_type_id:label'], 0, 3) === 'NSW') {
```

Comment:
OK so if they go via contact-qld to a NSW form, they get a QLD checkbox and not a NSW checkbox.  Is that acceptable?

Reply:
Ah fair point. I'll change the conditional test for the QLD domain too.